climada/engine/test/test_impact.py

```diff
@@ -816,6 +816,21 @@ def test__exp_build_event(self):
         self.assertEqual(exp.value_unit, imp.unit)
         self.assertEqual(exp.ref_year, 0)
 
+class TestAssignCentroids(unittest.TestCase):
+
+    def test_assign_centroids(self):
+        "Test that hazard centroids get assigned correctly"
+        exp = ENT.exposures
+        exp.assign_centroids(HAZ)
+        fake_eai_exp = np.arange(len(exp.gdf))
+        fake_at_event = np.arange(HAZ.size)
+        fake_aai_agg = np.sum(fake_eai_exp)
+        imp = Impact.from_eih(exp, ENT.impact_funcs, HAZ,
+                              fake_at_event, fake_eai_exp, fake_aai_agg)
+        imp.assign_centroids(HAZ)
+        imp_centr = getattr(imp,f'centr_{HAZ.tag.haz_type}')
+        np.testing.assert_array_equal(imp_centr,exp.gdf.centr_TC)
+
 
 class TestImpactH5IO(unittest.TestCase):
```

Comment:
I'd also make another test for the utility function that checks all three main cases:
- the centroids are a raster
- the centroids are not a raster
- the two crs don't match

Reply:
I'm not so familiar with best practices for testing.
I thought since the function is basically a wrapper around 2 functions (for ``u_coord.assign_grid_points`` and ``u_coord.assign_coordinates``) which both are tested, and it is called from exp.assign_centroids() and imp.assign_centroids() which also both are tested, it would be sufficient.

So would the idea be that I replicate the tests from exp.assign_centroids() for [points](https://github.com/CLIMADA-project/climada_python/blob/e26f900ca0e7e3ea627555e9345c5ae76f911428/climada/entity/exposures/test/test_base.py#L58) and [rasters ](https://github.com/CLIMADA-project/climada_python/blob/e26f900ca0e7e3ea627555e9345c5ae76f911428/climada/entity/exposures/test/test_base.py#L126) except that I directly use the u_coord.assign_centroids_to_gdf instead of the exp.assign_centroids() ?

Reply:
Ideally the actual points and rasters assignment tests from exposures rather would be rather _moved_ to the test_coordinates module, And in the exposures tests only the impact on the exposures object is tested, like: the gdf has a new/modified column now.

Reply:
I added the suggested tests [here](https://github.com/CLIMADA-project/climada_python/pull/602/commits/58056448120691e87d23a6a4da9567421d5cae3b). Is this suitable?

Reply:
😍 Thanks!